channels/serializers.py

```diff
@@ -37,3 +43,134 @@ def update(self, instance, validated_data):
             kwargs['channel_type'] = validated_data['subreddit_type']
 
         return api.update_channel(name=name, **kwargs)
+
+
+class WriteableSerializerMethodField(serializers.SerializerMethodField):
+    """
+    A SerializerMethodField which has been marked as not read_only so that submitted data passed validation.
+    The actual update is handled in PostSerializer.update(...).
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.read_only = False
+
+    def to_internal_value(self, data):
+        return data
+
+
+class PostSerializer(serializers.Serializer):
+    """Serializer for posts"""
+    url = WriteableSerializerMethodField(allow_null=True)
+    text = WriteableSerializerMethodField(allow_null=True)
+    title = serializers.CharField()
+    upvoted = WriteableSerializerMethodField()
+    downvoted = WriteableSerializerMethodField()
```

Comment:
The user is allowed to write them in the update though. What would the comment be for?